azure/functions/meta.py

```diff
@@ -16,16 +22,28 @@
 
 
 def is_iterable_type_annotation(annotation: object, pytype: object) -> bool:
-    is_iterable_anno = (
-        typing_inspect.is_generic_type(annotation)
-        and issubclass(typing_inspect.get_origin(annotation),
-                       collections.abc.Iterable)
-    )
+    """Since python 3.9, standard collection types are supported in type hint.
+    origin is the unsubscripted version of a type (eg. list, union, etc.).
+
+    If origin is not None, then the type annotation is a builtin or part of
+    the collections class.
+    """
+    origin = get_origin(annotation)
+    if sys.version_info >= (3, 9):
+        is_iterable_anno = (origin is not None
```

Comment:
Why is `typing_inspect.is_generic_type(annotation)` not included in >3.9?

Reply:
`is_generic_type` will fail for the `list` type -- this is a method we defined. `get_origin` expands on our definition of `is_generic_type` but includes the built-in collection types. `get_origin` supports the generic types and returns a non-None value where `is_generic_type` would return true